python/ray/data/tests/test_block_boundaries.py

```diff
@@ -0,0 +1,27 @@
+import numpy as np
+
+from ray.data.block import _get_block_boundaries
+
+
+def test_groupby_map_groups_get_block_boundaries():
+    indices = _get_block_boundaries(
+        block={
+            "x": np.array([1, 1, 2, 2, 3, 3]),
+            "y": np.array([1, 1, 2, 2, 3, 4]),
+        }
+    )
+
+    assert list(indices) == [0, 2, 4, 5, 6]
+
+    indices = _get_block_boundaries(
+        block={
+            "x": np.array([1, 1, 2, 2, 3, 3]),
+            "y": np.array(["a", "b", "a", "a", "b", "b"]),
```

Comment:
Please add some Nones into it as well (for sanity checking as we had issues with None handling perviously)

Reply:
Ya, make senses. taking a look. coz' the sort() from `dataset` actually does not accept everything, e.g., #48927

Reply:
This should be resolved by https://github.com/ray-project/ray/pull/48750